### PR TITLE
feat: treat OutputInstance in template literal as rule violation too

### PR DESCRIPTION
**Pull request by blakegong**
`OutputInstance` should be treated the same as `Output`. Please check the tests.

Currently, this is "good code":

```typescript
const myOutput: pulumi.OutputInstance<number> = pulumi.output([1, 2, 3])[0];
const myBadTemplate = `${myOutput}`
```

This PR would fix the above code into:

```typescript
const myOutput: pulumi.OutputInstance<number> = pulumi.output([1, 2, 3])[0];
const myBadTemplate = pulumi.interpolate`${myOutput}`
```

### CODE REVIEW

**Comment by kpitzen** [comment]
Hi @blakegong !  Thank you so much for your contribution - my only request here would be to break these into two separate linting rules - I'm fine with duplicating existing code in order to make each individual linting failure more clear.  Once that's done, I'm happy to get this merged in :) Thanks again!

File: README.md [hunk 1]
```diff
@@ -1,15 +1,19 @@
 # @pulumi/eslint-plugin
+
```
Comment (blakegong) [inline README.md:2]: These spacing changes are from `npm run format`. It actually also adds changes to some `.github/workflow/*.yml` files. I'm ignoring them in this PR cuz they are too unrelated.

**Comment by blakegong** [comment]
@kpitzen no problem! It's in a new `no-output-instance-in-template-literal` rule now. Please have a look!

**Comment by kpitzen** [comment]
This looks great @blakegong - thank you for your contribution!  I'll get it merged and a new version released. :)

**Review by kpitzen — APPROVED** [review]

**Comment by kpitzen** [comment]
FYI @blakegong - this is now published in version `0.2.0`.  Thank you again!

**Comment by blakegong** [comment]
Thank you @kpitzen!

**Pull request merged**